structured_experiments/prep_ex_01/image_preproc_exp_01.py

```diff
@@ -0,0 +1,160 @@
+import cv2 as cv
+import numpy as np
+
+
+class Image:
+
+    """
+    Class for submitted images. Contains functionality to load, save and begin
+    processing using common methods in opencv-python.
+    """
+
+    def __init__(self, uri, submission_id, color_space, image_matrix=None):
+        self.uri = uri
+        self.submission_id = submission_id
+        self.color_space = color_space
+        self.image_matrix = image_matrix
+
+    def load_image(self):
+        """
+        Loads the image as a numpy array using the image uri as string.
+        ----------
+        Arguments
+            self.uri : string representation of filepath to image for loading
+        ----------
+        returns:
+            image matrix stored in class instance
+        """
+        self.image_matrix = np.array(cv.imread(self.uri))
+        return self.image_matrix
+
+    def save_image(self, save_uri, image_to_save):
+        """
+        Writes the image as specified using the save_uri string. Uri must contain file extension.
+        ----------
+        Arguments
+            save_uri : string representation of path to intended save file
+            image_to_save : n dimensional numpy array containing image as pixel values
+        ----------
+        returns:
+            Void
+        """
+        self.image_matrix = image_to_save
+        cv.imwrite(save_uri, image_to_save)
+
+    def get_mean_pixel_value(self, scope='channelBGR', channel='None'):
+        """
+        Gets the mean pixel value for each channel based on scope provided. Current scope limited to entire image
+        in BGR colorspace.
+        ----------
+        Arguments
+            scope : indicates the space from which to get the mean value from; ie n dimensional pixels, all image values,etc
+            channel : which of the BGR channels to return the mean for
+        ----------
+        Returns
+            image matrix stored in class instance
+        """
+
+        if scope == 'image':
+            return self.image_matrix.mean(axis=(0, 1)).mean()
+        elif scope == 'channelBGR':
+            return self.image_matrix.mean(axis=0)
+        elif scope == 'channel' and channel is not None:
+            idx = ['B', 'G', 'R'].index(channel)
+            return self.image_matrix[idx].mean()
+
+    def contrast_brightness(self, alpha, beta):
+        """
+        Adjust the image contrast and brightness with the methodology below.
+
+        new_image = alpha * old_image + beta
+        alpha > 0 is the gain
+        beta is the bias
+        These are also both referred to as contrast and brightness
+        ----------
+        Arguments
+            alpha : unsigned 8 bit integer declaring the gain gain of the image

```

Comment:
Did you mean to write 'gain' 2x?

Reply:
Gain should have been once. This will be broken up into it's own PR as it was integrated here in error. Thank you for the heads up.